packages/flutter_tools/lib/src/build_system/targets/common.dart

```diff
@@ -288,6 +290,46 @@ class KernelSnapshot extends Target {
     }
   }
 
+  void _addFeatureFlagsToDartDefines(List<String> dartDefines) {
```

Comment:
TODO: Move this to `FlutterCommand` like the Flutter version information. The build system can't easily detect changes to environment variables, which causes feature flags set by environment variables to be stale until you `flutter clean`.

https://github.com/flutter/flutter/blob/e74b5da782026336824f30b9d2cb172267b0bdfa/packages/flutter_tools/lib/src/runner/flutter_command.dart#L1480-L1500